ml-agents/mlagents/trainers/agent_processor.py

```diff
@@ -258,3 +258,8 @@ def __init__(
             self.behavior_id
         )
         self.publish_trajectory_queue(self.trajectory_queue)
+
+    def set_environment_stats(self, env_stats: Dict[str, float]) -> None:
+        for stat_name, val in env_stats.items():
+            # full_stat = f"Environment/{stat_name}"
+            self.stats_reporter.set_stat(stat_name, val)
```

Comment:
Followup from the design review: we can pretty easily add averaging of multiple values by using `add_stat` here instead. I was thinking about:
* Adding on optional enum arg to AddStats() on C# for AggregationType
* Adding the AggregationType to the message
* using (name, AggregationType) as the dictionary key
* deciding between set_stat or add_stat here based on the AggregationType

Reply:
👍 
Almost want to say that the default behavior (without the enum) is `add_stat` just from the types of things MikeWise recorded (# of food eaten, etc.) and what kinds of things game developers are likely to record (score, coins collected, ranking, etc.). But hard to say.

Reply:
So do we still only want to keep worker 0 if the stats are being averaged? Seems like for that case, we'd want them all,

Reply:
Updated this to
```
for stat_name, (val, agg_type) in env_stats.items():
    if agg_type == StatsAggregationMethod.AVERAGE:
        self.stats_reporter.add_stat(stat_name, val)
    elif agg_type == StatsAggregationMethod.MOST_RECENT:
        # In order to prevent conflicts between multiple environments,
        # only stats from the first environment are recorded.
        if worker_id == 0:
            self.stats_reporter.set_stat(stat_name, val)
```
I think that does the least-surprising thing in each case. I guess we could have a third enum value like MOST_RECENT_PRIMARY_WORKER that checks the worker_id, and make MOST_RECENT always set, but I don't think those will get used much anyway.